core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/TermFilter.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2020 B2i Healthcare Pte Ltd, http://b2i.sg
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.b2international.snowowl.core.request;
+
+import java.io.Serializable;
+
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
+import com.b2international.commons.exceptions.BadRequestException;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+/**
+ * Represents a {@link Serializable} encapsulation of term search configuration.
+ * 
+ * @since 7.11
+ */
+public final class TermFilter implements Serializable {
+
+	private static final long serialVersionUID = 1L;
+	
+	@NotNull
+	private final String term;
+	
+	@Min(1)
+	private final Integer minShouldMatch;
+	
+	private final boolean fuzzy;
+	private final boolean exact;
+	private final boolean parsed;
+
+	public TermFilter(final String term, final Integer minShouldMatch, final boolean fuzzy, final boolean exact, final boolean parsed) {
+		this.term = term;
```

Comment:
Trim the incoming term and remove custom trimming from all concrete search request implementations.